server/resource_server/dto/dto.go

```diff
@@ -67,6 +67,17 @@ type ResetPasswordDTO struct {
 	NewPassword string `json:"new_password" validate:"required"`
 }
 
-type RegisterUserPrecheckDTO struct {
+type ResetPasswordPrecheckDTO struct {
 	Username string `json:"username" validate:"required"`
 }
+
+type RegisterUserPrecheckDTO struct {
+	Username string `json:"username" validate:"required,min=3,max=50"`
+}
+
+type ResetPasswordDTOV2 struct {
+	Username  string `json:"username" validate:"required,min=3,max=50"`
+	Signature []byte `json:"signature" validate:"required"`
+	StoredKey string `json:"stored_key" validation:"required"`
+	ServerKey string `json:"server_key" validation:"required"`
```

Comment:
You could also set minimal length of the stored key and server key to be 1 (to check that these values are not empty)